### PR TITLE
feat: Service details page read only

### PR DESCRIPTION
**Pull request by mohandast52**
## Proposed changes

- Readonly for the SVM service details should work as expected
- All the buttons for the SVM Service state should be disabled - so that no interaction occurs
- Also, the code has been refactored, hence marking this as a breaking change.
- The details pages should work as expected for components, agents, and services (L1 & other networks) including Solana read-only.

https://github.com/valory-xyz/autonolas-registry-frontend/assets/22061815/174e7d9b-2742-4a47-95b9-68ad0268c56b

### Testing in Goerli for service management

- went through all 5 steps and was able to proceed without any error, here is an example 👇 

https://github.com/valory-xyz/autonolas-registry-frontend/assets/22061815/f6fa5713-86d8-42b1-8164-9e0541b26209

unbond-step

## Types of changes

What types of changes does your code introduce to `autonolas-registry-frontend`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] My pull request is targeted against the main branch (left side), and I've started my branch from our main.
- [ ] I've ensured that linting and unit tests pass locally with my changes, and CI also passes successfully.
- [ ] I've included tests to validate the effectiveness of my fix or to demonstrate that my feature functions correctly.
- [ ] I've verified that my changes do not reduce code coverage.
- [x] I've added any necessary documentation as needed.
- [x] I've confirmed that any dependent changes have been merged and published in downstream modules.
- [ ] If the onload page involves contract interaction, I've ensured it is client-side rendered using next/dynamic.
- [ ] I've updated next.config.js header to include the new URL for secure app interaction

### CODE REVIEW

**Comment by DavidMinarsch** [comment]
Have you done any e2e testing on goerli to check this all still works fine on the eth side?

**Review by kupermind — DISMISSED** [review]

**Comment by oaksprout** [comment]
Testing:

- is it suitable to show an error notification here? i.e. is it actually an error?
image

File: yarn.lock
Comment (oaksprout) [inline yarn.lock:1]: updated fe library

File: util/constants.jsx [hunk 1]
```diff
@@ -30,12 +30,12 @@ export const SERVICE_STATE = {
 };
 
 export const SERVICE_STATE_KEY_MAP = {
-  nonExistent: 0,
-  preRegistration: 1,
-  activeRegistration: 2,
-  finishedRegistration: 3,
-  deployed: 4,
-  terminatedBonded: 5,
+  nonExistent: '0',
```
Comment (oaksprout) [inline util/constants.jsx:33]: Updates enum handling, not sure why

File: package.json
Comment (oaksprout) [inline package.json:1]: Update frontend library

File: tests/components/ListServices/details.test.jsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import {
   getAgentInstanceAndOperator,
   getServiceAgentInstances,
   getBonds,
-} from 'common-util/Details/ServiceState/utils';
+} from 'components/ListServices/ServiceState/utils';
```
Comment (oaksprout) [inline tests/components/ListServices/details.test.jsx:11]: Reorg file structure around service lists and details

File: components/ListServices/utils.jsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import {
 import { getServiceContract } from 'common-util/Contracts';
 import { convertStringToArray } from 'common-util/List/ListCommon';
 import { filterByOwner } from 'common-util/ContractUtils/myList';
-import { getTokenDetailsRequest } from 'common-util/Details/ServiceState/utils';
+import { getTokenDetailsRequest } from 'common-util/Details/utils';
```
Comment (oaksprout) [inline components/ListServices/utils.jsx:9]: reorg file structure

File: components/ListServices/details.jsx [hunk 4]
```diff
@@ -1,39 +1,85 @@
...
     <Details
       type="service"
       id={id}
-      getDetails={() => getServiceDetails(id)}
-      getHashes={() => getServiceHashes(id)}
-      getOwner={() => getServiceOwner(id)}
-      getTokenUri={() => getTokenUri(id)}
-      handleUpdate={() => router.push(`${links.UPDATE_SERVICE}/${id}`)}
-      onDependencyClick={(e) => router.push(`${links.AGENTS}/${e}`)}
+      getDetails={getDetails}
```
Comment (oaksprout) [inline components/ListServices/details.jsx:70]: Refactor prop handling for Details component

File: components/ListServices/details.jsx [hunk 5]
```diff
@@ -1,39 +1,85 @@
...
 
-  if (isSvm) {
+  if (hasNoSvmPublicKey) {
     return (
-      <Result
-        status="warning"
-        title="This page is not available for SVM services yet!"
-      />
+      <EmptyMessage>
+        Please connect a wallet that hold SOL to view this page.
```
Comment (oaksprout) [inline components/ListServices/details.jsx:61]:
```suggestion
        To view this page, connect a wallet that holds SOL
```

In general, best to lead this kind of message with the action the user is trying to take, and follow up with the action they need to take to resolve the issue.

File: components/ListServices/ServiceState/utils.jsx [hunk 6]
```diff
@@ -181,10 +139,36 @@ export const onActivateRegistration = async (account, id, deposit) => {
 };
 
 /* ----- step 2 functions ----- */
+/**
```
Comment (oaksprout) [inline components/ListServices/ServiceState/utils.jsx:142]: improve documentation for step 2

**Review by oaksprout — DISMISSED** [review]

**Review by oaksprout — APPROVED** [review]

**Comment by kupermind** [comment]
@mohandast52 I'm not seeing Solana Devnet in the selector (I've even tried to connect a devnet wallet), so I'm just seeing 3 mainnet services.
Screenshot 2024-01-18 at 4 51 42 PM

**Review by kupermind — APPROVED** [review]

**Pull request merged**